lib/simple_thread_pool.rb

```
@@ -27,29 +27,42 @@ def initialize(max_threads)
     active_threads = @lock.synchronize { @threads.select(&:alive?) }
-    active_threads.each(&:join)
+    active_threads.each do |thread|
+      thread.join
+    rescue Exception # standard:disable Lint/RescueException
+      # Joining a thread that died with an unhandled exception re-raises that
+      # exception; those are the block's responsibility to handle, so they are
+      # not propagated here. Exceptions raised on the calling thread itself
+      # (e.g. Interrupt) are re-raised. This heuristic has a narrow race: a
+      # signal arriving on the calling thread at the same moment the joined
+      # thread dies will be swallowed rather than re-raised.
+      raise if thread.alive?
```

RuboCop (1.88.2)

[warning] Avoid rescuing the `Exception` class. Perhaps you meant to rescue `StandardError`?

(Lint/RescueException)